projects/vdk-plugins/vdk-oracle/src/vdk/plugin/oracle/ingest_to_oracle.py

```diff
@@ -94,19 +97,37 @@ def _add_columns(self, table_name: str, payload: List[Dict[str, Any]]) -> None:
                     if sample_value is not None
                     else "VARCHAR2(255)"
                 )
-                column_defs.append(f"{col} {column_type}")
+                column_defs.append((col, column_type))
 
+            string_defs = [f"{col_def[0]} {col_def[1]}" for col_def in column_defs]
             alter_sql = (
-                f"ALTER TABLE {table_name.upper()} ADD ({', '.join(column_defs)})"
+                f"ALTER TABLE {table_name.upper()} ADD ({', '.join(string_defs)})"
             )
             self.cursor.execute(alter_sql)
-            self.column_cache[table_name.upper()].update(new_columns)
+            self.column_cache[table_name.upper()].update(column_defs)
 
     # TODO: https://github.com/vmware/versatile-data-kit/issues/2929
     # TODO: https://github.com/vmware/versatile-data-kit/issues/2930
-    def _cast_to_correct_type(self, value: Any) -> Any:
-        if type(value) is Decimal:
+    def _cast_to_correct_type(self, table: str, column: str, value: Any) -> Any:
+        def cast_string_to_type(t: str, v: str) -> Any:
+            if t == "FLOAT" or t == "DECIMAL":
+                return float(v)
+            if t == "NUMBER":
+                v = v.capitalize()
+                return bool(v) if v in ["True", "False"] else int(v)
+            if "TIMESTAMP" in t:
+                return datetime.datetime.strptime(v, "%Y-%m-%dT%H:%M:%S")
```

Comment:
Not sure if this should be configurable somehow.

Reply:
I was thinking the same but I am not sure whether this would really matter for the users, if not i think it is better to stay like this bc changing it would complicate the code and the usage

Reply:
Maybe we can leave it like this for now. This is the default oracle date time format. I'm more concerned if someone wants to pass in just time or just a date. Oracle supports this, but it gets complicated on our end in this case.